### PR TITLE
Implement GitHub action

### PR DESCRIPTION
**Pull request by harnel-tngn**
Implement GitHub action.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved type/method matching behavior with clearer not-found errors and an added runtime guard for an edge-case lookup.

* **Documentation**
  * Added XML documentation and enabled generation of library docs.

* **CI/CD**
  * Added automated build/test workflow and a publish workflow for packaging and releasing.

* **Chores**
  * Centralized project defaults, streamlined project configs, and added NuGet packaging metadata plus symbol/package outputs.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    - name: Execute unit tests
+      run: dotnet test
+
+    # Pack projects
+    - name: .NET Pack
+      run: dotnet pack --configuration Release --output ./artifacts
+      
+    # Upload nuget package
+    - name: Publish to NuGet
+      run: dotnet nuget push ./artifacts/*.nupkg --api-key ${{secrets.NUGET_API_KEY}} --source https://api.nuget.org/v3/index.json
```
[the remark this answers is not in the document]
Comment (harnel-tngn) [reply .github/workflows/publish.yml:39]: Failing when re-run already-published version is intentional. Workflow publish.yml should only success when publish is completed.
